ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+
+```mermaid
+flowchart TD
+    A[ACP-224 activated] --> B{Is ACP176FeeManager precompile active?}
+
+    B -- Yes --> C[Use targetGas from precompile storage]
+
+    B -- No --> D{Is gas-target set in node config?}
+    D -- Yes --> E[Use gas-target from node config]
+
+    D -- No --> F{Is targetGas set in chainConfig.acp176FeeConfig?}
+    F -- Yes --> G[Use targetGas from chain config]
+
+    F -- No --> H[Use default targetGas e.g. 2 million]
+```
+
+## Backwards Compatibility
+
+ACP-224 will require a network update in order to activate the new fee mechanism. Another activation will also be required to activate the new fee manager precompile. The activation of precompile should never occur before the activation of ACP-224 (the fee mechanism) since the precompile depends on ACP-224’s fee update logic to function correctly.
+
+Activation of ACP-224 mechanism will deactivate the prior fee mechanism and the prior fee manager precompile. This ensures that there is no ambiguity or overlap between legacy and new pricing logic In order to provide a configuration for existing networks, a network upgrade override for both activation time and ACP-176 configuration parameters will be introduced.
+
+These upgrades will be optional at the moment. However with introduction of ACP-194 (SAE), it will be required to activate this ACP; otherwise the network will not be able to use ACP-194.
+
+## Reference Implementation
+
+
+## Security Considerations
+
+Generally it has same security considerations as [ACP-176](https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/176-dynamic-evm-gas-limit-and-price-discovery-updates/README.md#security-considerations). Due to the nature of these parameters are dynamic, the risk here is generally more than ACP-176 as it allows for more misconfiguration.
+
+Potentially any misconfiguration of parameters could leave the network vulnerable to a DoS attack or over charge transactions. If `TargetGas` (`T`) is set too low, then the network might not be able to process bigger transactions. similarly if it is set too high, then validators might not be able to keep up in the processing of blocks.
+
+## Open Questions
+
+* Does it make sense dynamically changing the D and Q parameters? Currently they give each block builder the ability to adjust the value of $T$ on the rate of change of by roughly $\frac{1}{1024}$ of its current value.
+
+* Similarly does it make sense to dynamically change the `RMult` parameter? Currently it is set to 2, making `R = 2 * T` which means gas price can increase and decrease at the same rate.
+
+* Should we rather name the precompile as `ACP224FeeManager`?
+
+### Target gas in node configuration
+
+`gas-target` in node configuration (json config) will be disabled upon activation of the `ACP176FeeManager` precompile. Should the precompile rather have a function to enable/disable that option? This will
```

Comment:
Looks like this section still needs to be completed.

Reply:
this is indeed a "open question" lol